plugins/styleguide/app/controllers/styleguide/styleguide_controller.rb

```diff
@@ -6,8 +6,13 @@ class StyleguideController < ApplicationController
     skip_before_action :check_xhr
 
     def index
-      if !current_user || !current_user.in_any_groups?(SiteSetting.styleguide_allowed_groups_map)
-        raise Discourse::InvalidAccess.new
+      if !current_user &&
+           !SiteSetting.styleguide_allowed_groups_map.include?(Group::AUTO_GROUPS[:everyone])
+        return raise Discourse::NotFound
+      end
+
+      if current_user && !current_user.in_any_groups?(SiteSetting.styleguide_allowed_groups_map)
+        return raise Discourse::NotFound
```

Comment:
returning `InvalidAccess` here meant that the html page was still being rendered, with a 403 returned on the underlying json request. Which didn't block access to the styleguide. Returning 404 shows the user a 404 page.